src/rpc/db/function.c

```diff
@@ -247,17 +246,18 @@ static int db_function_typecheck(string apikey, string name,
     return (-1);
   }
 
-  if (((argc = db_function_get_argc(apikey, name)) < 0) ||
+  if (((argc = db_function_get_argc(pluginkey, name)) < 0) ||
       ((size_t)argc != (size_t)args->size)) {
     LOG_WARNING("Invalid argument count!");
     return (-1);
   }
 
-  reply = redisCommand(rc, "LRANGE %s:func:%s:args 0 %d", apikey.str, name.str,
-          argc);
+  reply = redisCommand(rc, "LRANGE %s:func:%s:args 0 %d", pluginkey,
+            name.str, argc);
 
   if (reply->type == REDIS_REPLY_ARRAY)
-    for (size_t j = reply->elements; j != 0; j--) {
+    for (size_t j = reply->elements, k = 0; j != 0; j--, k++) {
```

Comment:
Maybe @bontric is able to say some words to your first point. I am interested in this loop, too.